src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/common/concat_sdp.cpp

```diff
@@ -14,235 +15,6 @@ using namespace CPUTestUtils;
-        create_input(function->get_parameters()[2], targetInputStaticShapes[0], idx + 3.0f);
-        create_input(function->get_parameters()[3], targetInputStaticShapes[1], idx + 4.0f);
-        create_input(function->get_parameters()[4], ov::Shape{targetInputStaticShapes[0][0]}, idx + 0.0f);
-    }
-    void prepare() {
-        compile_model();
-        inferRequest = compiledModel.create_infer_request();
-        ASSERT_TRUE(inferRequest);
-    }
-    void reset() {
-        for (auto&& state : inferRequest.query_state()) {
-            state.reset();
-        }
-    }
-    std::vector<ov::Tensor> run_test(std::shared_ptr<ov::Model> model) {
-        function = model;
-        prepare();
-        std::vector<ov::Tensor> outputs;
-        int idx = 0;
-        for (auto&& shapes : targetStaticShapes) {
-            generate(idx++, shapes);
-            for (const auto& input : inputs) {
-                inferRequest.set_tensor(input.first, input.second);
-            }
-            inferRequest.infer();
-            auto outputTensor = inferRequest.get_output_tensor(0);
-            ov::Tensor copy{outputTensor.get_element_type(), outputTensor.get_shape()};
-            outputTensor.copy_to(copy);
-            outputs.push_back(copy);
-        }
-        reset();
-
-        return outputs;
-    }
-
-    bool hasShapeOf;
-};
-
-TEST_P(ConcatSDPTest, CompareWithRefs) {
-    SKIP_IF_CURRENT_TEST_IS_DISABLED();
-    auto actualOutputs = run_test(function);
-    if (!hasShapeOf) {
-        CheckNumberOfNodesWithType(compiledModel, "ScaledDotProductAttention", 1);
-        CheckNumberOfNodesWithType(compiledModel, "Concatenation", 0);
-        CheckNumberOfNodesWithType(compiledModel, "Reorder", 0);
-        CheckNumberOfNodesWithType(compiledModel, "Gather", 0);
-    } else {
-        // SimplifyGatherShapeOf will generate subgraph which contains Gather/Concat/Reorder, so could not check the number to confirm it's expected.
-        // W/ or w/o fusion the SDPA name is the same, but if fused the output number should be 3(data+pastk+pastv)
-        for (const auto& node : compiledModel.get_runtime_model()->get_ops()) {
-            if (node->get_rt_info().find(ov::exec_model_info::LAYER_TYPE)->second.as<std::string>() == "ScaledDotProductAttention") {
-                ASSERT_EQ(3, node->get_output_size()) << "ScaledDotProductAttention should be fused";
-            }
-        }
-    }
-    if (inType == ElementType::f16) {
-        configuration["INFERENCE_PRECISION_HINT"] = "f32";
-    }
-    auto expectedOutputs = run_test(functionRefs);
-    CheckNumberOfNodesWithType(compiledModel, "ScaledDotProductAttention", 0);
-    for (size_t i = 0; i < actualOutputs.size(); i++) {
-        ov::test::utils::compare(expectedOutputs[i], actualOutputs[i], abs_threshold, rel_threshold);
-    }
-}
-
 namespace {
 const std::vector<std::vector<InputShape>> inputShapes = {
```

Comment:
To avoid `inputShapes` duplication in common and arm cpp files you can create a method that returns this object and call this method in `common/concat_sdp.cpp` and `arm/concat_sdp.cpp`.
As an example, you can review methods defined in `classes\conv_concat.hpp` - `inputShapes2D`, `inputShapes3D`, etc. and follow the same approach.